components/script/webdriver_handlers.rs

```diff
@@ -348,15 +354,18 @@ unsafe fn jsval_to_webdriver_inner(
                 &HandleValueArray::empty(),
                 value.handle_mut(),
             ) {
-                jsval_to_webdriver_inner(cx, global_scope, value.handle(), seen)
+                match jsval_to_webdriver_inner(cx, global_scope, value.handle(), seen) {
+                    Ok(converted_item) => Ok(converted_item),
+                    err @ Err(_) => return err,
+                }
```

Comment:
https://github.com/servo/servo/actions/runs/14900760667/job/41852004184?pr=36908
```
error: this match expression is unnecessary
   --> components/script/webdriver_handlers.rs:357:17
    |
357 | /                 match jsval_to_webdriver_inner(cx, global_scope, value.handle(), seen) {
358 | |                     Ok(converted_item) => Ok(converted_item),
359 | |                     err @ Err(_) => return err,
360 | |                 }
    | |_________________^ help: replace it with: `jsval_to_webdriver_inner(cx, global_scope, value.handle(), seen)`
```
I think Clippy has bugs... The suggested way has totally different logic. With its suggestion, Step 5 would be run even with failure..
To get around it, I am thinking about 
```suggestion
                Ok(jsval_to_webdriver_inner(cx, global_scope, value.handle(), seen)?)
```

Reply:
I prefer this replacement over the match.